presto-main/src/main/java/com/facebook/presto/sql/planner/LogicalPlanner.java

```diff
@@ -207,6 +218,55 @@ private RelationPlan createExplainAnalyzePlan(Analysis analysis, Explain stateme
         return new RelationPlan(root, scope, ImmutableList.of(outputSymbol));
     }
 
+    private RelationPlan createAnalyzePlan(Analysis analysis, Analyze analyzeStatement)
+    {
+        TableHandle targetTable = analysis.getTableHandle(analyzeStatement.getTable());
+        Map<String, Object> analyzeProperties = metadata.getAnalyzePropertyManager().getProperties(
+                targetTable.getConnectorId(),
+                targetTable.getConnectorId().getCatalogName(),
+                analysis.getAnalyzeTableProperties(),
+                session,
+                metadata,
+                analysis.getParameters());
+
+        AnalyzeTableMetadata analyzeTableMetadata = metadata.getAnalyzeTableMetadata(session, targetTable, analyzeProperties);
+        // Replace the target table with the one from analyze table metadata. The handle returned by getAnalyzeTableMetadata()
+        // may contain some additional information required for ANALYZE
+        targetTable = analyzeTableMetadata.getTableHandle();
```

Comment:
wouldn't it be better to require that any additional information should be included in some dedicated type?
I mean: the API forces connector implementation to sometimes include certain information in the TableHandle and sometimes not -- so upon accepting ConnectorTableHandle, they cannot really know if this information should be present; This is messy, it would be better, if there was some support from the type system for this. Maybe `ConnectorAnalyzeTableMetadata` should be an interface, included verbatim in `AnalyzeTableMetadata`?

Reply:
It complicates things a lot. Currently we encode this information in TableHandle, so the `getTableLauyout` can process it, and return correct partitions.

Reply:
We had some discussion around this topic and the conclusion was the current one. Under the current codebase, we considered 2 options:
(1) (current implementation) Encode the scan ranges in the table handle and when `AddExchange` pass the handle to connector for table layouts, connector constraint the layout on the scan ranges. 
(2) Encode the scan ranges as filter predicates in the plan and rely on it to aggregate on the desired ranges. It's less obvious and has assumptions on predicate push down. 

Otherwise, we need to change the engine to add a different channel for the connector to influence the scan range, which complicate things a lot. I personally think that the table handle approach is a good fit under the current codebase.